### ISSUE TITLE
Rule proposal: switch-case-limits

### ISSUE THREAD

**Issue by lo1tuma** [sindresorhus/eslint-plugin-unicorn#2439]
### Description

A rule that enforces the min or max amount of cases required per switch statement. E.g. a switch statement with just one case or two would be better expressed with `if-else`. At the same time a switch statement with too many cases might be better refactored to a lookup table.

```json
{
  "switch-case-limits": [
    "error",
    { "min": 2, "max": 10 }
  ]
}
```

### Fail

```js
// min: 2
switch (foo) {
   case: 'bar':
      return 'BAR';
}
```

```js
// max: 3
switch (foo) {
   case: 'bar1':
      return 'BAR1';
   case: 'bar2':
      return 'BAR2';
   case: 'bar3':
      return 'BAR3';
   case: 'bar4':
      return 'BAR4';
}
```

### Pass

```js
// min: 2
switch (foo) {
   case: 'bar1':
      return 'BAR1';
  case: 'bar2':
      return 'BAR2';
}
```

```js
// max: 3
switch (foo) {
   case: 'bar1':
      return 'BAR1';
   case: 'bar2':
      return 'BAR2';
   case: 'bar3':
      return 'BAR3';
}
```

### Proposed rule name

switch-case-limits

**Comment by fregante** [comment]
I agree that `switches` with one `case` unnecessarily verbose, I don't think a large number of cases is inherently bad though. 

If they're a list of 15 case/return lines line in your example I think it's fine. Many cases are usually unreadable when each case has 30+ lines each. At that point the native `complexity` or `max-lines-per-function` might just do it.

**Comment by sindresorhus** [comment]
I also don't really see the problem with large switches. I do think the first example with the single case could be its own rule though.

**Comment by sindresorhus** [comment]
We are passing on this.
